### PR TITLE
Java 21 migration and UI changes 

### PR DESCRIPTION
**Pull request by dgurjar-ch**
Java 21 migration and UI changes
Upgrade to java 21 and springboot 3
Update header for changes
Update footer for new logo and copyright

### CODE REVIEW

**Review by foman-ch — APPROVED** [review]

File: src/test/java/uk/gov/companieshouse/web/lfp/controller/lfp/EnterLFPDetailsControllerTest.java [hunk 1]
```diff
@@ -128,8 +128,6 @@ void postRequestCompanyNumberLowerCase() throws Exception {
                 .param(PENALTY_NUMBER_ATTRIBUTE, VALID_PENALTY_NUMBER)
                 .param(COMPANY_NUMBER_ATTRIBUTE, LOWER_CASE_LLP))
                 .andExpect(status().is3xxRedirection())
-                .andExpect(model().attributeExists(TEMPLATE_NAME_MODEL_ATTR))
```
Comment (dhawes-ch) [inline src/test/java/uk/gov/companieshouse/web/lfp/controller/lfp/EnterLFPDetailsControllerTest.java:131]: These attribute checks have been removed and I notice it's after a redirect. Are you sure we can remove these, and that we don't instead need to update the controllers to use [flash attributes](https://docs.spring.io/spring-framework/reference/web/webmvc/mvc-controller/ann-methods/flash-attributes.html) ?

Comment (dgurjar-ch) [reply src/test/java/uk/gov/companieshouse/web/lfp/controller/lfp/EnterLFPDetailsControllerTest.java:131]: Yes , we need these attribute and I have added again and with change in controller to add flush attributes. Thank you

**Review by dhawes-ch — APPROVED** [review]
Looks much better

File: src/main/java/uk/gov/companieshouse/web/lfp/controller/lfp/EnterLFPDetailsController.java [hunk 2]
```diff
@@ -46,6 +47,9 @@ public class EnterLFPDetailsController extends BaseController {
     private static final String LFP_ONLINE_PAYMENT_UNAVAILABLE = "/online-payment-unavailable";
 
     private static final String PENALTY_TYPE = "penalty";
+    private static final String TEMPLATE_NAME_MODEL_ATTR = "templateName";
+    private static final String ENTER_LFP_DETAILS_MODEL_ATTR = "enterLFPDetails";
+    private static final String BACK_BUTTON_MODEL_ATTR = "backButton";
```
Comment (dhawes-ch) [inline src/main/java/uk/gov/companieshouse/web/lfp/controller/lfp/EnterLFPDetailsController.java:52]: This should be acquired from the baseController method addBackPageAttributeToModel

**Review by dhawes-ch — APPROVED** [review]

**Pull request merged**